common/esxi_download_datastore_file.yml

```diff
@@ -0,0 +1,34 @@
+# Copyright 2022 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Download file from ESXi datastore
+# Parameters:
+#   src_datastore: the datastore name of the file. e.g. datastore1
+#   src_file_path: the relative file path in datastore. e.g. vm_name/vmware.log
+#   dest_file_path: the downloaded file path at localhost. e.g. /tmp/downloaded_vmware.log
+#   download_file_timeout: timeout in seconds for downloading datastore file. Default is 300s.
+
+- include_tasks: esxi_check_delete_datastore_file.yml
+  vars:
+    file_in_datastore_ops: file
+    file_in_datastore: "{{ src_datastore }}"
+    file_in_datastore_path: "{{ src_file_path }}"
+    file_in_datastore_failed_ignore: False
+
+- name: "Set fact of downloading URL for the src datastore file"
+  set_fact:
+    datastore_file_url: "{{ ds_file_result.url }}"
```

Comment:
`ds_file_result.url` can be checked if it's defined or `datastore_file_url` will be set to "".

Reply:
Added an assert to check ds_file_result.url before set_fact.